brainiak/utils/fmrisim.py

```diff
@@ -2134,126 +2660,229 @@ def generate_noise(dimensions,
+    mean_signal = (base[mask > 0]).mean()
+
     # Generate the noise
     noise_temporal = _generate_noise_temporal(stimfunction_tr=stimfunction_tr,
                                               tr_duration=tr_duration,
                                               dimensions=dimensions,
                                               template=template,
                                               mask=mask,
-                                              fwhm=noise_dict[
-                                                 'fwhm'],
-                                              motion_sigma=noise_dict[
-                                                 'motion_sigma'],
-                                              drift_sigma=noise_dict[
-                                                 'drift_sigma'],
-                                              auto_reg_sigma=noise_dict[
-                                                 'auto_reg_sigma'],
-                                              physiological_sigma=noise_dict[
-                                                 'physiological_sigma'],
+                                              noise_dict=noise_dict,
                                               )
 
-    # Create the base (this inverts the process to make the template)
-    base = template * noise_dict['max_activity']
-
-    # What is the mean signal of the non masked voxels in this template?
-    mean_signal = (base[mask > 0]).mean()
-
     # Convert SFNR into the size of the standard deviation of temporal
     # variability
     temporal_sd = (mean_signal / noise_dict['sfnr'])
 
-    # Calculate the sd that is necessary to be combined with itself in order
-    # to generate the temporal_sd
-    temporal_sd_element = np.sqrt(temporal_sd ** 2 / 2)
+    # Calculate the temporal sd of the system noise (as opposed to the noise
+    #  attributed to the functional variability).
+    temporal_sd_system = np.sqrt((temporal_sd ** 2) * temporal_proportion)
 
     # What is the standard deviation of the background activity
-    spatial_sd = mean_signal / noise_dict['snr']
+    spat_sd = mean_signal / noise_dict['snr']
+    spatial_sd = np.sqrt((spat_sd ** 2) * (1 - temporal_proportion))
 
     # Set up the machine noise
     noise_system = _generate_noise_system(dimensions_tr=dimensions_tr,
                                           spatial_sd=spatial_sd,
-                                          temporal_sd=temporal_sd_element,
+                                          temporal_sd=temporal_sd_system,
                                           )
 
-    # Reshape the base (to be the same size as the volume to be created)
-    base = base.reshape(dimensions[0], dimensions[1], dimensions[2], 1)
-    base = np.ones(dimensions_tr) * base
-
     # Sum up the noise of the brain
-    noise = base + (noise_temporal * temporal_sd_element) + noise_system
+    noise = base + (noise_temporal * (1 - temporal_sd)) + noise_system
```

Comment:
I may be wrong, but I am not sure if (1 - temporal_sd) is what you intended.

Reply:
I think this code is out of date (github doesn't make it easy to track comments). The current code reads: noise = base + (noise_temporal * temporal_sd) + noise_system

Reply:
I see. sorry then maybe I went to the old code...